drivers/modem/hl7800.c

```diff
@@ -1783,7 +1783,13 @@ static bool on_cmd_atcmdinfo_iccid(struct net_buf **buf, uint16_t len)
 		LOG_INF("EID: %s", delim + 1);
 	}
 
-	ret = snprintk(iface_ctx.mdm_iccid, sizeof(iface_ctx.mdm_iccid), "%s", value);
+	ret = strlen(value);
+	strncpy(iface_ctx.mdm_iccid, value, sizeof(iface_ctx.mdm_iccid));
+	if (ret > (sizeof(iface_ctx.mdm_iccid) - 1)) {
+		ret = sizeof(iface_ctx.mdm_iccid) - 1;
+	}
+	iface_ctx.mdm_iccid[ret] = '\0';
+
 	if (ret > MDM_HL7800_ICCID_MAX_STRLEN) {
 		LOG_WRN("ICCID too long: %d", ret);
```

Comment:
Now this will not print the original `value` string length anymore, but the truncated length that fits into `iface_ctx.mdm_iccid`. Is that intended?

`snprintf` returns the length that would have been written if the buffer was large enough.

Reply:
Good catch. This is what we want, but then I believe the `LOG_WRN` is now pointless.

Reply:
also can `ret` be `> MDM_HL7800_ICCID_MAX_STRLEN`? (aka `MDM_HL7800_ICCID_MAX_SIZE - 1`)? I think this needs few more tweaks :-)

Reply:
ok pushed one more variant, now logging both the original and truncated string length